lib/Sema/PlaygroundTransform.cpp

```diff
@@ -813,14 +823,38 @@ class Instrumenter : InstrumenterBase {
 
     llvm::SmallVector<Expr *, 6> ArgsWithSourceRange(Args.begin(), Args.end());
 
-    ArgsWithSourceRange.append(
-        {StartLine, EndLine, StartColumn, EndColumn, ModuleExpr, FileExpr});
-
     UnresolvedDeclRefExpr *LoggerRef = new (Context)
         UnresolvedDeclRefExpr(LoggerName, DeclRefKind::Ordinary,
                               DeclNameLoc(SR.End));
     LoggerRef->setImplicit(true);
 
+    if (ExtendedCallbacks) {
+      auto SourceLocBufferId = Context.SourceMgr.findBufferContainingLoc(SR.Start);
+      StringRef filePath;
+
+      // Use #sourceLocation if it exists.
+      SourceLoc sl = SR.Start;
+      auto *vf = Context.SourceMgr.getVirtualFile(sl);
```

Comment:
SourceManager has an operation `getDisplayNameForLoc` that is used for printing the name in other compiler outputs. Would you prefer to use that?

Reply:
Thanks for the suggestion @DougGregor!  In this case, what we'd want to pass to the callback is the actual path through which the file can later be opened — the phrase "display name" suggests that this function might not always return a literal path, but from the implementation, it looks as if it does.  Ideally we'd want the same path that is emitted in `.dia` files — from the code, it looks as if that's indeed using `getDisplayNameForLoc` — is that correct?

Reply:
I believe that's correct

Reply:
Thanks for the advice.  It looks good as far as I can tell.  I'll update the PR.